### PR TITLE
[Azure Pipelines] Run infrastructure/ tests on Windows 10

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

**Comment by foolip** [comment]
Getting Python installed is the first challenge. I've tried `msiexec /a python-2.7.15.amd64.msi` but it fails with "The Windows Installer Service could not be accessed. This can occur if the Windows Installer is not correctly installed. Contact your support personnel for assistance." @thejohnjansen do you know the best way to install stuff? Does the context where the Azure Pipelines steps are run have administrator rights, or some equivalent of sudo rights?

**Comment by foolip** [comment]
@thejohnjansen this is looking quite promising, I got Python installed and got the tools/ unittests running, although as expected with failures.

Trying to run the infrastructure/ tests with Edge I got stuck and not being able to update System32\drivers\etc\hosts. I see these options:
- Set up everything that requires administrator access as part of the VM setup
- Make it possible to run steps as adminstrator in Azure Pipelines. It looks like that currently isn't possible: https://github.com/Microsoft/azure-pipelines-tasks/issues/5073

We could make the former work, but it seems more fragile than having full control over the VM in `.azure-pipelines.yml` and rolling back to a clean state between every run.

**Comment by foolip** [comment]
Hmm, this will be tricky. We have to run as administrator by default to do a lot of the setup, but can't run `MicrosoftWebDriver`as adminstrator. But in order to do `./wpt run` as non-admin, the whole checkout will also have to be done as the user, and any file that the non-admin user writes may have to be copied by the admin user to the artifacts directory. It's all probably doable, but pretty intricate.

**Comment by foolip** [comment]
https://github.com/Microsoft/azure-pipelines-tasks/issues/5073 would help.

**Comment by foolip** [comment]
Oh no. https://serverfault.com/questions/773038/run-as-a-different-user-without-a-password-from-an-elevated-prompt-in-windows looks like it's going to be a problem. We need either password-less `sudo` or password-less "anti-sudo", but right now it doesn't seem like either exist :(

File: .azure-pipelines.yml
Comment (gsnedders) [inline .azure-pipelines.yml]: `s/10 Windows/Windows 10/`

File: .azure-pipelines.yml
Comment (gsnedders) [inline .azure-pipelines.yml]: Why are we getting rid of the macOS jobs here?

File: .azure-pipelines.yml
Comment (gsnedders) [inline .azure-pipelines.yml]: We don't install Firefox above?

**Comment by foolip** [comment]
This is all still an experiment. I've self-assigned, but if I remove reviewers I suspect wpt-pr-bot will just add them back. Sorry :/

**Comment by gsnedders** [comment]
@foolip I was mostly having a quick look with regards to running the unit tests on Win Server!

File: .azure-pipelines.yml
Comment (foolip) [reply .azure-pipelines.yml]: Just to not trigger them for every test commit and waste resources.

File: .azure-pipelines.yml
Comment (foolip) [reply .azure-pipelines.yml]: We use `--install-browser` here.

File: .azure-pipelines.yml
Comment (foolip) [inline .azure-pipelines.yml]: This is the last thing that should be uncommented before merging this.

**Comment by foolip** [comment]
This is ready for real review now. @gsnedders?

**Comment by foolip** [comment]
After discussion with @thejohnjansen, renaming to focus on Windows 10 stable and not Insider Preview.

File: .azure-pipelines.yml
Comment (gsnedders) [inline .azure-pipelines.yml]: Why is this not in a YML file like the hosts update?

Comment (foolip) [reply .azure-pipelines.yml]: I don't like this bit, hoping to not need it when this is merged. That's blocked on the agent pool being reliably populated with responsive agents that don't get their state messed up so easily.

Comment (gsnedders) [reply .azure-pipelines.yml]: What else is this waiting on now? You've got the review you asked for now. :)

Comment (foolip) [reply .azure-pipelines.yml]: I'll write email about the agent pool.

**Comment by foolip** [comment]
@gsnedders this is now failing only because /infrastructure/reftest/size.html fails. I'll file a bug and mark that as an expected failure.

**Comment by foolip** [comment]
@gsnedders this is ready for final review or even merging (squashing!) now.

**Review by Ms2ger — APPROVED** [review]
Please squash!

**Comment by foolip** [comment]
Squahed! Thanks @Ms2ger!

**Pull request merged**
